src/realm/obj.cpp

```diff
@@ -2363,9 +2416,22 @@ void Obj::assign_pk_and_backlinks(const Obj& other)
         }
         void on_mixed_property(ColKey col) final
         {
-            REALM_ASSERT(m_origin_obj.get_any(col).is_null() ||
-                         m_origin_obj.get_any(col).get_link().get_obj_key() == m_dest_orig.get_key());
-            m_origin_obj.set(col, Mixed{m_dest_replace.get_link()});
+            auto val = m_origin_obj.get_any(col);
+            if (val.is_type(type_TypedLink)) {
```

Comment:
this code doesn't handle if val is null anymore, isn't that still a possibility?

Reply:
Good catch!!!

Reply:
On the other hand: If the original value was NULL there would never be a backlink to handle. @ironage do you agree?